### PR TITLE
fix: share with dialog ui

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-4260 mf-4259 mf-4262 mf-4258 mf-4261 mf-4257

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx [hunk 1]
```diff
@@ -105,7 +116,29 @@ export function SelectProfileUI(props: SelectProfileUIProps) {
...
+                'linkedTwitterNames',
+            ],
+            isCaseSensitive: false,
+            ignoreLocation: true,
+            threshold: 0,
+        })
+            .search(keyword)
+            .map((item) => item.item)
+            .filter((x) => !frozenPubkeyList.includes(x.linkedPersona?.publicKeyAsHex!))
+    }, [keyword, items, frozenPubkeyList])
```
Comment (UncleBill) [inline packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx]: Do not recreate a Fuse each time the keyword changes.

```ts
const fuse = useMemo(() => {
return Fuse.create(items, ...)
}, [items])

const results = useMemo(() => {
  return fuse.search(keyword)
}, [fuse, keyword]

```

By the way, `frozenPubkeyList` should be memorized above

```diff
-    const frozenPubkeyList = compact(frozenSelected.map((x) => x.linkedPersona?.publicKeyAsHex))
+    const frozenPubkeyList = useMemo(() => compact(frozenSelected.map((x) => x.linkedPersona?.publicKeyAsHex)), [frozenSelected])
```

File: packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx [hunk 2]
```diff
@@ -120,52 +153,65 @@ export function SelectProfileUI(props: SelectProfileUIProps) {
...
                         [],
                     )}
-                    onKeyDown={(event) => {
-                        if (search !== '' || event.key !== 'Backspace') return
-                        onSetSelected(selected.slice(0, selected.length - 1))
+                    onKeyDown={(e) => {
+                        if (e.code !== 'Enter') return
+
+                        // onSetSelected(selected.slice(0, selected.length - 1))
+                        startTransition(() => props.onSearch(search))
```
Comment (UncleBill) [inline packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx]: Don't forget to remove commented code.

File: packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx [hunk 3]
```diff
@@ -120,52 +153,65 @@ export function SelectProfileUI(props: SelectProfileUIProps) {
...
-                                    />
-                                )
-                            })
-                        )}
-                    </Box>
+            {props.loading ? (
+                <div className={cx(classes.empty, classes.mainText)}>
+                    <LoadingBase />
+                    <Typography>{t('loading')}</Typography>
                 </div>
```
Comment (UncleBill) [inline packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx]: We also have a `<LoadingStatus />` component.

File: .vscode/settings.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
         "editor.defaultFormatter": "esbenp.prettier-vscode"
     },
     "[typescriptreact]": {
-        "editor.defaultFormatter": "esbenp.prettier-vscode"
+        "editor.defaultFormatter": "vscode.typescript-language-features"
```
Comment (guanbinrui) [inline .vscode/settings.json]: Please revert it.

**Review by guanbinrui — APPROVED** [review]

**Comment by guanbinrui** [comment]
Need reconfirmation from @UncleBill

**Review by UncleBill — APPROVED** [review]

File: packages/mask/src/components/InjectedComponents/SelectPeopleDialog.tsx [hunk 5]
```diff
@@ -60,35 +65,63 @@ const useStyles = makeStyles()((theme) => ({
...
+    const type = resolveNextIDPlatform(valueToSearch)
+    const value = resolveValueToSearch(valueToSearch)
+    const { loading: searchLoading, value: NextIDResults } = usePersonasFromNextID(
+        value,
+        type ?? NextIDPlatform.NextID,
+        MaskMessages.events.ownProofChanged,
+        false,
+    )
+
+    const NextIDItems = useTwitterIdByWalletSearch(NextIDResults, value, type)
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/SelectPeopleDialog.tsx:97]: This component is used in all platforms. Why you use a Twitter-specific hook in it?

**Pull request merged**